internal/data/LocalSpiceDbContainer.go

```diff
@@ -0,0 +1,198 @@
+package data
+
+import (
+	"ciam-rebac/internal/conf"
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/authzed/authzed-go/v1"
+	"github.com/go-kratos/kratos/v2/log"
+	"io"
+	"path"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/ory/dockertest"
+)
+
+const (
+	// SpicedbImage is the image used for containerized spiceDB in tests
+	SpicedbImage = "authzed/spicedb"
+	// SpicedbVersion is the image version used for containerized spiceDB in tests
+	SpicedbVersion = "v1.22.2"
+	// SpicedbSchemaBootstrapFile specifies an optional bootstrap schema file to be used for testing
+	SpicedbSchemaBootstrapFile = "spicedb-test-data/basic_schema.yaml"
+	// SpicedbRelationsBookstrapFile specifies an optional bootstrap file containing relations to be used for testing
+	SpicedbRelationsBookstrapFile = ""
+)
+
+// LocalSpiceDbContainer struct that holds pointers to the container, dockertest pool and exposes the port
+type LocalSpiceDbContainer struct {
+	logger    log.Logger
+	port      string
+	container *dockertest.Resource
+	pool      *dockertest.Pool
+}
+
+// CreateContainer creates a new SpiceDbContainer using dockertest
+func CreateContainer(logger log.Logger) (*LocalSpiceDbContainer, error) {
+	pool, err := dockertest.NewPool("") // Empty string uses default docker env
+	if err != nil {
+		return nil, fmt.Errorf("could not connect to docker: %w", err)
+	}
+
+	pool.MaxWait = 3 * time.Minute
+
+	var (
+		_, b, _, _ = runtime.Caller(0)
+		basepath   = filepath.Dir(b)
+	)
+
+	cmd := []string{"serve-testing", "--skip-release-check=true"}
+
+	var mounts []string
+	if SpicedbSchemaBootstrapFile != "" {
+		cmd = append(cmd, "--load-configs")
+		cmd = append(cmd, "/mnt/spicedb_bootstrap.yaml")
+		mounts = append(mounts, path.Join(basepath, SpicedbSchemaBootstrapFile)+":/mnt/spicedb_bootstrap.yaml")
+	}
+	if SpicedbRelationsBookstrapFile != "" {
```

Comment:
I wonder how long this was "bookstrap" - I tried to do a proposed edit but I don't think I can